packages/vite/src/node/server/sourcemap.ts

```diff
@@ -8,6 +8,12 @@ const debug = createDebugger('vite:sourcemap', {
   onlyWhenFocused: true
 })
 
+// Virtual modules should be prefixed with a null byte to avoid a
+// false positive "missing source" warning. We also check for colons,
+// because the optimizer emits sourcemaps with sources like "dep:react"
+// due to namespacing for special handling in esbuildDepPlugin.
+const virtualSourceRE = /:|\0/
```

Comment:
I would prefer this check to be more targeted. What about 
```suggestion
const virtualSourceRE = /^(\0|dep:)/
```